core/chains/evm/gas/chain_specific.go

```diff
@@ -7,7 +7,7 @@ import (
 
 // chainSpecificIsUsable allows for additional logic specific to a particular
 // Config that determines whether a transaction should be used for gas estimation
-func chainSpecificIsUsable(tx evmtypes.Transaction, cfg Config) bool {
+func chainSpecificIsUsable(tx evmtypes.Transaction, block evmtypes.Block, cfg Config) bool {
```

Comment:
It looks like only `block.BaseFeePerGas` field is used. What about passing that value in directly instead?
```suggestion
func chainSpecificIsUsable(tx evmtypes.Transaction, baseFeePerGas *big.Int, cfg Config) bool {
```

Reply:
Why do you say that is more generic with the block type?

The Go idiom here is to accept only what you need to in order to maximize generality (i.e. to be able to call it without constructing an extra block when we don't already have one). We could still make the parameter a generic type either way later on.

Reply:
Ya, makes sense to just use baseFeePerGas for now.
In future if you need other things, can edit it later.